### PR TITLE
Change names: Ingressor, Egressor, [type]Link

### PR DESCRIPTION
**Pull request by CollinValley**
Ben Weis is having some trouble getting this done on his macbook due to the tests not running for some reason. Something due to not enough file descriptors to run 20+ instances of tokio at once.  So I just went ahead with his naming suggestion and implemented the concept. Added him as a reviewer.  I also removed unnecessary Element words from things.

### CODE REVIEW

File: route-rs-graphgen/src/main.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ use std::collections::HashMap;
 mod codegen;
 mod pipeline_graph;
 
-enum ElementLink {
+enum SyncLink {
```
Comment (scgruber) [inline route-rs-graphgen/src/main.rs]: This actually should be generic, I just don't have other cases yet.

Comment (CollinValley) [reply route-rs-graphgen/src/main.rs]: Okay, should the name be changed back? Or should it just be Link?

Comment (scgruber) [reply route-rs-graphgen/src/main.rs]: It could just be `Link`, if that doesn't feel too terse.

**Review by ddowl — APPROVED** [review]
Wow, these PRs are getting big. Happy with these new names, they're much more clear

**Pull request merged**